src/cmd/cli/command/compose.go

```diff
@@ -89,6 +90,13 @@ func makeComposeUpCmd() *cobra.Command {
 			tailCtx, cancelTail := context.WithCancelCause(cmd.Context())
 			defer cancelTail(nil) // to cancel WaitServiceState and clean-up context
 
+			ctx := tailCtx
+			if !detach && waitTimeout >= 0 {
```

Comment:
`detach` is already checked on line 85, so the `!detach` case never happens here